packages/abstract-sdk/src/apis/distribution.rs

```diff
@@ -0,0 +1,147 @@
+//! # Distribution
+//! Interacts with the distribution module of cosmos
+//!
+
+use cosmos_sdk_proto::{
+    cosmos::{base, distribution},
+    traits::Message,
+};
+use cosmwasm_std::{to_binary, Addr, Coin, CosmosMsg};
+
+use crate::{
+    features::{AbstractNameService, AccountIdentification},
+    AbstractSdkResult,
+};
+
+pub trait DistributionInterface: AbstractNameService + AccountIdentification {
```

Comment:
We have these trait bounds here but we're not using any of these traits in the implementation. 

Ideally we already construct the execution message to be called on the proxy contract (see the other apis)